### PR TITLE
Misc fixes to DynComp

### PR DESCRIPTION
**Pull request by markro49**

### CODE REVIEW

File: java/daikon/dcomp/Premain.java [hunk 1]
```diff
@@ -70,50 +70,13 @@
   protected static Set<String> problem_classes =
       new HashSet<>(
           Arrays.asList(
-              // Interfaces marked @FunctionalInterface should not be instrumented
-              // as they are used by LambdaExpressions.  Since BCEL won't let us
-              // inspect the annotations, we must manually add to problem classes.
-              "java.util.regex.Pattern$CharPredicate",
```
Comment (markro49) [inline java/daikon/dcomp/Premain.java:76]: The work I did to get JUnit tests working also fixed this.

File: java/daikon/dcomp/Premain.java [hunk 2]
```diff
@@ -70,50 +70,13 @@
...
-              "java.lang.Class.getDeclaredClasses",
-              "java.lang.Class.getDeclaredField",
-              "java.lang.Class.getDeclaredFields",
-              "java.lang.Class.getDeclaredMethod",
-              "java.lang.Class.getDeclaredMethods",
-              "java.lang.Class.getDeclaredConstructor",
-              "java.lang.Class.getDeclaredConstructors",
-              "java.util.ResourceBundle.getBundle",
-              "java.util.ResourceBundle.clearCache",
-
```
Comment (markro49) [inline java/daikon/dcomp/Premain.java:114]: The work I did to get @HotSpotCandidate working also fixed @CallerSensitive.

File: java/daikon/dcomp/Premain.java [hunk 3]
```diff
@@ -70,50 +70,13 @@
...
-              // The below entries are temporary, until the bugs are fixed.
```
Comment (mernst) [inline java/daikon/dcomp/Premain.java:115]: Should this be retained?

**Review by mernst — APPROVED** [review]

File: java/daikon/dcomp/Premain.java (hunk 3 above)
Comment (markro49) [reply java/daikon/dcomp/Premain.java:115]: It turns out there appear to be no problem methods now - I deleted rest of list.

Comment (mernst) [reply java/daikon/dcomp/Premain.java:115]: Wow, that's great!

**Pull request merged**